packages/framework/collabspaces/src/reverseMap.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export type ReverseMapType = "row" | "col";
+
+export interface IReverseMap {
+	getRowId(rowId: string): number | undefined;
+	getColId(colId: string): number | undefined;
+
+	getRowMap(): { [id: string]: number };
+	getColMap(): { [id: string]: number };
+
+	removeCellsFromMap(type: ReverseMapType, start: number, count: number): void;
+
+	addCellToMap(type: ReverseMapType, id: string, index: number): void;
+}
+
+export class ReverseMap implements IReverseMap {
+	private readonly rowMap: { [id: string]: number } = {};
+	private readonly colMap: { [id: string]: number } = {};
+
+	constructor() {}
```

Comment:
Maybe I'm missing something, but this should fail as you do not populate initial state after loading matrix.
It would work only when we create it from scratch, not when we load existing matrix.

Reply:
whenever any cell gets changed/added the callbacks populate it. So we do not need to do anything here.